packages/docusaurus-theme-classic/src/theme/MDXComponents/index.tsx

```diff
@@ -31,6 +32,7 @@ const MDXComponents: MDXComponentsObject = {
   h4: (props) => <MDXHeading as="h4" {...props} />,
   h5: (props) => <MDXHeading as="h5" {...props} />,
   h6: (props) => <MDXHeading as="h6" {...props} />,
+  admonition: Admonition,
```

Comment:
not directly related to this PR but we could add tab components as well

I'd eventually like a theme / site to be able to more easily register mdx shortcodes

Reply:
@slorber See #6370. We just need each theme to be able to enhance the previous one's `MDXComponents`, but that requires fixing https://github.com/facebook/docusaurus/issues/4530, which then requires us to figure out what JS loader we want to use, because it involves JS transformation, as we talked about on Discord.

Reply:
@Josh-Cena this is not really the approach I'd like to take on this (multiple layers of theme mdx providers)

What I'd prefer is to emit a new file in `.docusaurus`


Users/plugins would just provide a mapping, usable in other contexts. This mapping could easily compose and look like:


```js
{
  mdx: {
    admonition: "@theme/Admonition"
    tab: "@theme/Tab"
    tabs: "@theme/Tabs"
  },
  navbar: {
    colorPicker: "@site/src/components/Navbar/ColorPicker",
    login: "@site/src/components/Navbar/Login",
  },
  footer: {
    xyz: "@site/src/components/Navbar/Xyz",
  } 
```


And the emitted file (in `.docusaurus`) would us `React.lazy(() => import("@theme/Xyz"))`

I guess we first need to upgrade to React 18 and ensure React.lazy + SSR works well before

Reply:
@slorber Themes enhance the `MDXComponents`, not providers. It would be:

```js
import OriginalMDXComponents from "@theme-original/MDXComponents";

export default {
  ...OriginalMDXComponents,
  element: Component,
};
```

Reply:
@Josh-Cena I understand that but at the same time this may not be easy to implement + produce a more awkward API for the end-user.

If you want to add custom navbar/footer/mdx items to a site you'd have to swizzle 3 components for example.

And we still want to use dynamic imports + `React.lazy()` on this to benefit from better code splitting and selective hydration. A user that does not use a given item type on a page should not necessarily need to download that code.

Reply:
What's a navbar/footer MDX item? 😅

Dynamic imports make sense. In fact we can do that today. Just not sure of the benefit: our wrapper components are extremely lightweight, and users are encouraged to have lightweight mappers as well.

I'm not opposed to having a `@generated/mdxComponents` file. In fact that's exactly what I proposed in #6370. However I realized we can do that with existing APIs so there's no need to make a new one. I'm still trying to make the core minimally aware of the existence of MDX.

Reply:
> What's a navbar/footer MDX item? 😅

custom item types, not necessarily mdx (also needed for docs sidebar, blog sidebar etc...)

> Dynamic imports make sense. In fact we can do that today. 

Those won't work on the server/SSR by default unless we wrap them with React-Loadable or React.lazy + Suspense.

This is probably not the best place to discuss all this 😅 just wanted to mention that we could add tabs